example/app/AppWrapperSync.tsx

```diff
@@ -0,0 +1,112 @@
+import React, {useRef, useState} from 'react';
+import {Realm} from '@realm/react';
+import {Pressable, SafeAreaView, StyleSheet, Text} from 'react-native';
+
+import TaskContext from './models/Task';
+import {LoginScreen, AuthState} from './components/LoginScreen';
+import colors from './styles/colors';
+import {AppSync} from './AppSync';
+import {buttonStyles} from './styles/button';
+import {shadows} from './styles/shadows';
+
+export const AppWrapperSync: React.FC<{
+  appId: string;
+}> = ({appId}) => {
+  const {RealmProvider} = TaskContext;
+
+  // Set up the Realm app
+  const app = useRef(new Realm.App({id: appId})).current;
+
+  // Store the logged in user in state so that we know when to render the login screen and
+  // when to render the app. This will be null the first time you start the app, but on future
+  // startups, the logged in user will persist.
+  const [user, setUser] = useState<Realm.User | null>(app.currentUser);
+
+  const [authState, setAuthState] = useState(AuthState.None);
+  const [authVisible, setAuthVisible] = useState(false);
+
+  // If the user presses "login" from the auth screen, try to log them in
+  // with the supplied credentials
+  const handleLogin = async (email: string, password: string) => {
```

Comment:
Is it on purpose we're not using `useCallback` to memoize these arrow functions?

Reply:
Oversight.  Going to change that.  Good catch